components/runtime/erlang/Makefile

```diff
@@ -23,71 +23,55 @@
-COMPONENT_NAME=     erlang
-COMPONENT_VERSION=    24.2.2
-COMPONENT_PROJECT_URL=  http://www.erlang.org/
+COMPONENT_FMRI= runtime/erlang
+COMPONENT_NAME= erlang
+COMPONENT_VERSION= 26.2.2
+COMPONENT_PROJECT_URL= https://erlang.org/
 COMPONENT_SRC_NAME= otp_src
-COMPONENT_SRC=    $(COMPONENT_SRC_NAME)_$(COMPONENT_VERSION)
-COMPONENT_ARCHIVE= OTP-${COMPONENT_VERSION}.tar.gz
+COMPONENT_SRC= $(COMPONENT_SRC_NAME)_$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE= ${COMPONENT_SRC}.tar.gz
 COMPONENT_ARCHIVE_HASH= \
-	sha256:b6adfc0bf14d94348146ae26cc38d09dca545f8e14ebab7ddcf9482a6e8d1162
-COMPONENT_ARCHIVE_URL=  https://github.com/erlang/otp/archive/${COMPONENT_ARCHIVE}
-COMPONENT_POST_UNPACK_ACTION= ( $(MV) otp-OTP-$(COMPONENT_VERSION) $(COMPONENT_SRC))
-COMPONENT_PREP_ACTION = (cd $(@D); ./otp_build update_configure --no-commit)
-COMPONENT_VERSION_SHORT= $(shell echo $(COMPONENT_VERSION) | cut -d'.' -f1-2)
+	sha256:d537ff4ac5d8c1cb507aedaf7198fc1f155ea8aa65a8d83edb35c2802763cc28
+COMPONENT_ARCHIVE_URL= https://github.com/erlang/otp/releases/download/OTP-${COMPONENT_VERSION}/${COMPONENT_ARCHIVE}
+COMPONENT_POST_UNPACK_ACTION= ( $(RM) -r $(COMPONENT_SRC)/lib/wx )
+COMPONENT_VERSION_SHORT= 26.2
 COMPONENT_CLASSIFICATION= Development/Other Languages
-COMPONENT_LICENSE=        Apache-2.0
-COMPONENT_LICENSE_FILE=   LICENSE.txt
+COMPONENT_LICENSE= Apache-2.0
+COMPONENT_LICENSE_FILE= LICENSE.txt
 COMPONENT_SUMMARY= Erlang/OTP
 COMPONENT_DESCRIPTION= Erlang programming language and OTP libraries
 
 # manpages
 COMPONENT_NAME_1= otp_doc_man
-COMPONENT_SRC_1=  $(COMPONENT_NAME_1)_$(COMPONENT_VERSION_SHORT)
-COMPONENT_ARCHIVE_1=  $(COMPONENT_SRC_1).tar.gz
+COMPONENT_SRC_1= $(COMPONENT_NAME_1)_26.2.2
+COMPONENT_ARCHIVE_1= $(COMPONENT_SRC_1).tar.gz
 COMPONENT_ARCHIVE_HASH_1= \
-    sha256:56ae4b13fc7c003f4dd9951a9ff2a0bae0b38473df2c078d79c8553c277c5f4c
-COMPONENT_ARCHIVE_URL_1=  $(COMPONENT_PROJECT_URL)download/$(COMPONENT_ARCHIVE_1)
-UNPACK_ARGS_1=  -r $(COMPONENT_SRC_1)
-PKG_PROTO_DIRS += $(COMPONENT_SRC_1)
-
-# HTML documentation
-COMPONENT_NAME_2= otp_doc_html
-COMPONENT_SRC_2=  $(COMPONENT_NAME_2)_$(COMPONENT_VERSION_SHORT)
-COMPONENT_ARCHIVE_2=  $(COMPONENT_SRC_2).tar.gz
-COMPONENT_ARCHIVE_HASH_2= \
-    sha256:f479cbc8a28532fd6a0a55fc26684b4e79312da4f86ee0735d0757f936672bbc
-COMPONENT_ARCHIVE_URL_2=  $(COMPONENT_PROJECT_URL)download/$(COMPONENT_ARCHIVE_2)
-UNPACK_ARGS_2=  -r $(COMPONENT_SRC_2)
-PKG_PROTO_DIRS += $(COMPONENT_SRC_2)
+    sha256:16c1c6a949d619bf63b5c2fe9872dc043322e1130416595f22f0bc4c70459f47
+COMPONENT_ARCHIVE_URL_1= https://github.com/erlang/otp/releases/download/OTP-26.2.2/$(COMPONENT_ARCHIVE_1)
+UNPACK_ARGS_1= -r $(COMPONENT_SRC_1)
 
 include $(WS_MAKE_RULES)/common.mk
 
 SYSTEM_TEST_TARGET= configure $(SYSTEM_TEST_64)
 
-CFLAGS += -DOPENSSL_NO_MD4 -DOPENSSL_NO_RC2 -DOPENSSL_NO_RC4
-
 # Fix 64-bit linking.
 LDFLAGS += $(CC_BITS)
+CFLAGS += -finline-functions -funroll-loops
+CXXFLAGS += -finline-functions -funroll-loops -fno-exceptions -fno-rtti
```

Comment:
There is some background to this. illumos uses a forked version of GCC and we are still partially not upstreamed. It's a lot better but some things need testing when it comes to commandline options as they have in the past not exactly behaved as expected. However we thus far have found all problems and fixed them in the compiler. So if we are not using these flags and most other distributions (including linx and bsd) do we should start using them. Although we need to gather some packaging experience with them.

Reply:
> So if we are not using these flags and most other distributions (including linx and bsd) do we should start using them.

Okay, let's start to gather some information:
* Fedora Linux does *not* build `erlang` with `-fno-rtti`: https://src.fedoraproject.org/rpms/erlang/blob/rawhide/f/erlang.spec#_759
* nixpkgs uses `-fno-rtti` in four packages (out of 90k+): https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20fno-rtti&type=code

I'd really appreciate some reference to a repository with the massive usage of the `-fno-rtti` option.

Reply:
* Solaris - zero occurrences of `-fno-rtti` in their `solaris-userland`: https://github.com/search?q=repo%3Aoracle%2Fsolaris-userland%20fno-rtti&type=code

Reply:
> ... are the default for games and things like the mozilla browsers for good reason.

If some particular project (like a game or web browser) is confident they are safe to be built using `-fno-rtti` then they can do so internally and that's perfectly okay because the particular project knows its internals best.  They can also take some measurements to make sure their internal usage of `-fno-rtti` is still correct.  If they found it is no longer okay to be build with `-fno-rtti` they can just drop it.

I believe we build some packages with `-fno-rtti` and we do not know that just because this is hidden somewhere in their `Makefile` forrest (or whatever) and we simply do not care about project's internals too much.

But I consider too risky to force `-fno-rtti` from an outside of the project and ... hope?  or believe?  ... it will work properly.  And it will do so forever.  There are missing test results for `erlang` so we do not know the actual situation and we cannot evaluate the impact of added `-fno-rtti`.  And even with 100% passed test results the justification of added `-fno-rtti` is still weak.

Reply:
> > Re `-finline-functions`: since we build with `-O2` by default (and I believe it is same for erlang too) this option should not be needed to be set explicitly.
> 
> true. in practive however it turned out that, depending on os/platform/gcc version, it's not as reliable as having that option explicitly. and since it doesn't hurt having it in addition ...

I disagree with this approach[*], but I can live with it because added `-finline-functions` is effectively no-op.  If `-finline-functions` is a must for you, then please add a comment saying something similar as you wrote above or anything else explaining why you think it is needed.

[*] if we would accept this is the way how to do things then somebody else could come and add `-fvar-tracking` with the exactly same explanation, then somebody else will add `-fshrink-wrap`, etc.  Where it will end?  Makefiles will be full of unnecessary stuff making maintenance of components harder.

> > Re `-fno-exceptions`: this is to disable some compile warnings.
> 
> no.

Yes, you are right.  I messed this with `-Wno-exceptions`.  I'm sorry.

The `-fno-exceptions` option falls under the same category as `-fno-rtti`, i.e. I do not think it is safe to force it from outside of a project.

BTW, if both `-fno-rtti` and `-fno-exceptions` are so good (and safe) for `erlang` then why not to propose their addition directly [into the `erlang` project](https://github.com/erlang/otp)?  This would make `erlang` better not only for OpenIndiana, but also for everybody else.

Reply:
Ok, I have now looked through the compiler options in detail and had some discussions off the list with @RAC-oi these options give benefits and since compilation works, we save on C++ features which erlang does not use. All small improvements are good for small projects. Endless discussions just add frustration and are not helpful. As stated before the userbase for erlang on OpenIndina is small. We don't even have the most popular consumer namely rabbitmq packaged. So even if it where risky it's a good testbed. Marking as resolved and moving on.